### ISSUE TITLE
delagent clogging scheduler queue

### ISSUE THREAD

**Issue by yaobinshi** [fossology/fossology#296]
---

Author Name: **Bob Gobeille** 
Original Redmine Issue: 6939, http://www.fossology.org/issues/6939
Original Date: 2014/04/21
Original Assignee: larry shi

---

Having delagent run in exclusive mode introduces a serious productivity problem.  Here is an example I just ran into:

1) long running upload (3.5M files) is partially done (running nomos).
2) a small delagent job queued
3) three more jobs queued after delagent

In order to get the jobs in step 3 started, the user canceled the delagent.  However, the scheduler will still not start the jobs in step 3.  I manually deleted the job records from the database and told the scheduler to check the job queue.  But the scheduler still won't start the step 3 jobs.  Doing a scheduler status shows that the scheduler still knows about delagent and it is still blocking the subsequent jobs even though it is marked as JOB_FAILED.  Changing the priority of a job in the scheduler also seems to have no effect.

To summarize:

1) A deleted job that is EXCLUSIVE, should not stop the queue from processing subsequent jobs
2) Scheduler job priorities should work

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **Bob Gobeille** 
Original Date: 2014/10/06 20:49:18 +0000

---

Just ran into this again on fossology.fc.  This is a major problem because the only fall back is to manually delete the delagent job from the database.

**Comment by mcjaeger** [comment]
requires test

**Comment by mcjaeger** [comment]
actually, on 'Version: [3.4.0], Branch: [master], Commit: [#6ad6a1]'

* uploading larger package (to have some nomos runtime)
* delete a package
* schedule a new job

-> upload gets deleted, but scheduler is crashed. and subsequent jobs are not executed We have some other PR here that was merged after version 3.4: 

https://github.com/fossology/fossology/pull/1258

-> still needs test
